Add tt_entails test case

field845 opened the pull request:
Add test_tt_entails() on test_entail_engine.py

field845 committed:
Add tt_entails test case

Add test_tt_entails() on test_entail_engine.py

field845 committed:
Update test_entail_engine.py

field845 committed:
implemented fc

roughly done fc and test cases, will do the refactor a bit later

field845 committed:
changed test file name

field845 committed:
Delete test_tt_entails1.txt

field845 committed:
Delete test_tt_entails2.txt

field845 committed:
Delete test_tt_entails3.txt

field845 committed:
Delete test_tt_entails4.txt

field845 committed:
Refactor code

field845 committed:
format the fc print statement

field845 committed:
Refactor Code

John-Iliadis reviewed (changes requested):
- main() is in main.py. Remove redefinition and delete iengine.py
- all the knowledge base classes should be in one file
- all the entail algorithms should in one file
- never import *
- .txt should be in the data folder
- remove logic.ipynb
- in expr.py, what is the purpose of is_prop_symbol?

field845 committed:
fixed bugs and changed is_prop_symbol function

field845 commented:
Hi, is_prop_symbol() is the start point of the forward chaining algorithm

field845 committed:
refactor the code

put all the knowledge base classes in one file
put all the entail algorithms in one file
change import *
remove logic.ipynb

field845 committed:
modify fc symbol print statement

field845 committed:
finish bc

field845 committed:
Delete logic.ipynb

field845 committed:
Delete main.py

field845 committed:
delete print debug statement from bc

field845 committed:
Add test cases
